Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -1194,6 +1211,81 @@ theorem map_sub [DecidableEq ι] (m : ∀ i, M₁ i) (i : ι) (x y : M₁ i) :
   rw [sub_eq_add_neg, sub_eq_add_neg, MultilinearMap.map_add, map_neg]
 #align multilinear_map.map_sub MultilinearMap.map_sub
 
+
+lemma map_update [DecidableEq ι] (x : (i : ι) → M₁ i) (i : ι) (v : M₁ i)  :
+    f (update x i v) = f x - f (update x i (x i - v)) := by
+  rw [map_sub, update_eq_self, sub_sub_cancel]
+
+open Finset in
+lemma map_sub_map_piecewise [LinearOrder ι] (a b : (i : ι) → M₁ i) (s : Finset ι) :
+    f a - f (s.piecewise b a) =
+    ∑ i in s, f (fun j ↦ if j ∈ s → j < i then a j else if i = j then a j - b j else b j) := by
+  refine s.induction_on_min ?_ fun k s hk ih ↦ ?_
+  · rw [Finset.piecewise_empty, sum_empty, sub_self]
+  rw [Finset.piecewise_insert, map_update, ← sub_add, ih,
+      add_comm, sum_insert (lt_irrefl _ <| hk k ·)]
+  simp_rw [s.mem_insert]
+  congr 1
+  · congr; ext i; split_ifs with h₁ h₂
+    · rw [update_noteq, Finset.piecewise_eq_of_not_mem]
+      · exact fun h ↦ (hk i h).not_lt (h₁ <| .inr h)
+      · exact fun h ↦ (h₁ <| .inl h).ne h
+    · cases h₂
+      rw [update_same, s.piecewise_eq_of_not_mem _ _ (lt_irrefl _ <| hk k ·)]
+    · push_neg at h₁
+      rw [update_noteq (Ne.symm h₂), s.piecewise_eq_of_mem _ _ (h₁.1.resolve_left <| Ne.symm h₂)]
+  · apply sum_congr rfl; intro i hi; congr; ext j; congr 1; apply propext
+    simp_rw [imp_iff_not_or, not_or]; apply or_congr_left'
+    intro h; rw [and_iff_right]; rintro rfl; exact h (hk i hi)
+
+/-- This calculates the differences between the values of a multilinear map at
+two arguments that differ on a finset `s` of `ι` of cardinality `n`. It requires a
+linear order on `ι` in order to express the result.-/
+lemma map_piecewise_sub_map_piecewise [LinearOrder ι] (a b v : (i : ι) → M₁ i) (s : Finset ι) :
+    f (s.piecewise a v) - f (s.piecewise b v) = ∑ i in s, f
+      fun j ↦ if j ∈ s then if j < i then a j else if j = i then a j - b j else b j else v j := by
```

Comment:
This is probably a good place to us `ltbyCases`

Reply:
Is it? You'd have to write `fun _ ↦ a j` etc. which doesn't look clean ...